### PR TITLE
Proof of concept language server that lists S3 buckets

### PR DESCRIPTION
**Pull request by awschristou**
This change adds a concept language server that provides S3 bucket names as completion suggestions. It works on files that are named `*.s3.json`

This language server is being submitted so that we can perform additional experimentation like providing language servers with credentials, verifying browser compatibility, etc.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lsp/tsconfig.json [hunk 1]
```diff
@@ -30,11 +30,17 @@
...
+            "path": "./server/aws-lsp-s3"
+        },
         {
             "path": "./app/aws-lsp-buildspec-binary"
         },
         {
             "path": "./app/aws-lsp-cloudformation-binary"
+        },
+        {
+            "path": "./app/aws-lsp-s3-binary"
```
Comment (justinmk3) [inline lsp/tsconfig.json:43]: Looks like many projects. And the size of the package-lock.json in this PR is a hint about the dependency explosion we will need to manage with all of these micro-projects.

**Review by justinmk3 — APPROVED** [review]
A canonical skeleton/scaffold is very useful, nice.

File: lsp/tsconfig.json (hunk 1 above)
Comment (awschristou) [reply lsp/tsconfig.json:43]: The cool thing about how an npm monorepo works is that there is a single package-lock.json (and single node_modules) folder that serves as the composite for all projects in the monorepo. Each project still uses what it needs (without bundling the other packages that it doesn't reference), but you don't have redundant node_modules folders for each project.

Comment (justinmk3) [reply lsp/tsconfig.json:43]: Nice!

**Pull request merged**